maxprocs/maxprocs_test.go

```diff
@@ -141,3 +141,20 @@ func TestSet(t *testing.T) {
 		assert.Equal(t, 42, currentMaxProcs(), "should change GOMAXPROCS to match quota")
 	})
 }
+
+func TestMain(m *testing.M) {
+	prev, ok := os.LookupEnv(_maxProcsKey)
+	if err := os.Unsetenv(_maxProcsKey); err != nil {
+		fmt.Printf("Couldn't clear %s: %v\n", _maxProcsKey, err)
+		os.Exit(1)
+	}
+
+	code := m.Run()
+	if ok {
+		if err := os.Setenv(_maxProcsKey, prev); err != nil {
```

Comment:
I'm not sure restoring is important here, since we have an `os.Exit` right below